docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,98 @@
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
+
+### Implement the UI in a new Cell type
+
+This option will create a new nbextension to serve a new cell type as an extension to the code cell.
+
+* `+` This would mean working with a fresh codebase without existing constraints.
+* `+` Intermediate, minimally functional changes, can be rapidly prototyped and released.
+* `-` To stick with the DRY principle, this would mean refactoring parts of the current app cell to make various components importable (message bus responses, job state management, tab menu, etc.) instead of their integration (really we might want to consider that anyway).
+* `-` Much of the structure of the mockup already functions similar to the existing App Cell and might become duplicated work with additional maintenance burden.
+* `-` This would mean updating other components that make use of Narratives - the Static Narrative interface, Search, and the Dashboard, for example, all introspect Narratives and handle cells based on their type.
+
+### Don't use a Cell, use some other interface
+
+The bulk importer UI work has been mocked up as a cell that becomes a part of the Narrative workflow, but that could change. This option would mean using some other interface rather than a cell to implement bulk import work. 
+
+This has been discussed before, especially in the context of how Narratives are intended to work as a repeatable record of analysis and discussion. Generally, importing files to data objects isn't very useful in trying to repeat work, especially if you don't have access to those files. This will also happen when those files expire on the server and are no longer available. In that case, including one or more import cells is misleading - those cells cannot be run again, implying that the analysis that the Narrative performs cannot be completely reproduced.
+
+* `+` Unimpeded by the constraints around the cell architecture as imposed by both the Jupyter Notebook and KBase codebases.
+* `+` Keeping all import jobs consolidated in one place, outside of cells, would reduce Narrative length and Narrative object size.
+* `+` Removing import cells into a separate interface would have a positive impact on load time.
+* `-` Entrypoint to the interface would be something entirely new.
+* `-` New work would need to be done to make use of the app state management systems and kernel communication.
```

Comment:
```
New work would need to be done to make use of the app state management systems and kernel communication.
```
I could see  this done completely using the frontend job clients only, ignoring the need for kernel communication

Reply:
I could, too, but it makes me nervous to have some cells maintained only on the front end, and some cells maintained by the kernel. Kernel-side operations also make it easier to control data flow between the front end and the EE2 service. State lookups are only a single RPC that happens in a loop, and I think it would be best to keep it that way as much as possible.

Reply:
Since we're sticking with cells as they are with their code submission to start jobs, we can stick with the existing management architecture.

Reply:
(from meeting 10/28 9:30am PT)